iothub/device/tests/Authentication/IotHubConnectionStringBuilderTests.cs

```diff
@@ -23,27 +22,106 @@ public class IotHubConnectionStringBuilderTests
         private const string ModuleId = "moduleId";
         private const string SharedAccessKey = "dGVzdFN0cmluZzE=";
         private const string SharedAccessKeyName = "AllAccessKey";
-        private const string CredentialScope = "Device";
-        private const string CredentialType = "SharedAccessSignature";
         private const string SharedAccessSignature = "SharedAccessSignature sr=dh%3a%2f%2facme.azure-devices.net&sig=dGVzdFN0cmluZzU=&se=87824124985&skn=AllAccessKey";
+#pragma warning disable SYSLIB0026 // Type or member is obsolete
+        private static readonly X509Certificate2 s_cert = new();
+#pragma warning restore SYSLIB0026 // Type or member is obsolete
+        private static readonly X509Certificate2Collection s_certs = new();
 
         [TestMethod]
         public void IotHubConnectionStringBuilder_ParamConnectionString_ParsesHostName()
         {
-            var connectionString = $"HostName={HostName};SharedAccessKeyName={SharedAccessKeyName};DeviceId={DeviceId};SharedAccessKey={SharedAccessKey}";
+            string connectionString = $"HostName={HostName};SharedAccessKeyName={SharedAccessKeyName};DeviceId={DeviceId};SharedAccessKey={SharedAccessKey}";
             var iotHubConnectionCredentials = new IotHubConnectionCredentials(connectionString);
             iotHubConnectionCredentials.IotHubHostName.Should().Be(HostName);
             iotHubConnectionCredentials.HostName.Should().Be(HostName);
         }
 
+        [TestMethod]
+        public void IotHubConnectionStringBuilder_ParamConnectionString_ValidateToString()
+        {
+            string connectionString = $"HostName={HostName};DeviceId={DeviceId};ModuleId={ModuleId};SharedAccessKeyName={SharedAccessKeyName};SharedAccessKey={SharedAccessKey}";
+            IotHubConnectionString connString = IotHubConnectionStringParser.Parse(connectionString);
+            connString.ToString().Should().BeEquivalentTo(connectionString);
+        }
+
+        [TestMethod]
+        public void IotHubConnectionStringBuilder_ParamConnectionString_ValidateHostName()
+        {
+            // missing hostname
```

Comment:
If you updated the test name, we wouldn't need the comment.

```suggestion
        public void IotHubConnectionStringBuilder_ParamConnectionString_MissingHostNameThrows()
        {
```